crates/optimism/rpc/src/eth/mod.rs

```diff
@@ -80,7 +90,11 @@ where
     >,
 {
     /// Creates a new instance for given context.
-    pub fn new(ctx: &EthApiBuilderCtx<N>, sequencer_http: Option<String>) -> Self {
+    pub fn new(
+        ctx: &EthApiBuilderCtx<N>,
+        sequencer_http: Option<String>,
+        storage_proof_only: Vec<Address>,
+    ) -> Self {
```

Comment:
I have a feeling we might need more settings in the future so we should change the way we configure this with some builder types for `OpEthApi` and `OpAddons`

like 

```rust
let api =  OpEthApi::builder(ctx).with_sequencer().proof_only().build()
```

same for OpAddons